### ISSUE TITLE
false positive "Tuple index out of range" when indexing `tuple[Any, *tuple[Any, ...]]`

### ISSUE THREAD

**Issue by jorenham** [python/mypy#19858]
```py
from typing import Any

def f0n(xs: tuple[Any, ...]) -> int:
    return xs[42]  # ✔️ 

def f1n(xs: tuple[int, *tuple[Any, ...]]) -> int:
    return xs[42]  # ❌ 
```

```
main.py:7: error: Tuple index out of range  [misc]
main.py:7: note: Variadic tuple can have length 1
```

https://mypy-play.net/?mypy=latest&python=3.13&gist=9dc847117579110c1406d6b1cf3f2b35

this occurs with 1.17.1, 1.18.1, and master

possibly related to #19109

**Comment by ilevkivskyi** [comment]
As you can see from the note this is intentional. For historical reasons plain `tuple[X, ...]` is handled in unsafe manner. But types that can be only expressed with unpack syntax are handled strictly.

**Comment by ilevkivskyi** [comment]
Oh, I guess you want `Any` to be special-cased, if yes, then we already have an issue for this IIRC.

**Comment by ilevkivskyi** [comment]
https://github.com/python/mypy/issues/18665

**Comment by jorenham** [comment]
Hm no that's not what I meant. You see, the issue here is that `tuple[Any, ...]` is the "gradual tuple", which is not only "any-isch" in value, but also in size. That is, `tuple[Any, ...]` is assignable to every other tuple type *and vice-versa*. That's why mypy correctly (as can be seen in the example) accepts indexing it for any index.

A restricted form of the gradual tuple is `tuple[int, *tuple[Any, ...]]`, which has the additional requirement that 1) the type at index 0 is `int` and 2) that the size must be at least 1 (not 0). Just like before, it should be allowed to arbitrarily index it, since it's gradual in size. So it's length is a a hypothetical gradual positive `int` of at least `1`, i.e. `Literal[1] | (PosInt & Any)`.
But mypy does not behave that way here. Instead, it reports that indexing it beyond `1` can be type-unsafe. Technically this is true. But by gradual construction, we have indicated that it should be treated *as if* it has any size.

The issue you linked is certainly related, but I don't think its the same thing here. This is about the special-cased tuple type itself, not about user-defined variadically-gradual types.
